pegasus/retry_failover.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package pegasus
+
+import (
+	"context"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type tableRPCOp func() (confUpdated bool, result interface{}, err error)
+
+// retryFailOver retries the operation when it encounters replica fail-over, until context reaches deadline.
+func retryFailOver(ctx context.Context, op tableRPCOp) (interface{}, error) {
+	bf := backoff.NewExponentialBackOff()
+	bf.InitialInterval = time.Second
```

Comment:
`InitialInterval` means the first retry interval is 1 second since the last failure？

Reply:
Yes.

Reply:
So if user timeout set <1s, it will not be retried though it's `ERR_OBJECT_FOUND` or `ERR_INVALID_STATE`?

Reply:
Currently yes. It's a protection of our service, otherwise, the clients will concurrently query config in a very short period.

To be clear, `InitialInterval` is not a deterministic value, the first retry interval is `randomFactor * initialInterval`, at the minimum it's 500ms.

Reply:
So in `user timeout <500ms` case , the retry logic is invalid which never retry , I think, it should no be our expectation.